jerry-core/ecma/builtin-objects/ecma-builtin-math.c

```diff
@@ -153,6 +155,79 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
   return ecma_make_number_value (result_num);
 } /* ecma_builtin_math_object_max_min */
 
+#if ENABLED (JERRY_ES2015_BUILTIN)
+
+/**
+ * The Number.prototype object's 'Math.trunc' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 20.2.2.35
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_number_t
+ecma_builtin_math_object_trunc (ecma_number_t arg)
+{
+  if (ecma_number_is_nan (arg)||ecma_number_is_infinity (arg)||ecma_number_is_zero (arg))
+  {
+    return arg;
+  }
+
+  if ((arg > 0) && (arg < 1))
+  {
+    return (ecma_number_t) 0;
+  }
+
+  if ((arg < 0) && (arg > -1))
+  {
+    return (ecma_number_t) -0;
+  }
+
+  return (ecma_number_t) arg - fmod (arg,1);
+} /* ecma_builtin_math_object_trunc */
+
+#endif /* ENABLED (JERRY_ES2015_BUILTIN) */
+
+#if ENABLED (JERRY_ES2015_BUILTIN)
+
+/**
+ * The Number.prototype object's 'Math.sign' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 20.2.2.29
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_number_t
+ecma_builtin_math_object_sign (ecma_number_t arg)
+{
+  if (ecma_number_is_nan (arg))
+  {
+    return arg;
+  }
+
+  if (ecma_number_is_negative (arg))
+  {
+    if (ecma_number_is_zero (arg))
```

Comment:
Would it be possible to merge the positive and negative `is_zero` checks and return with `arg`? Or even merge it into the `is_nan` check.